sql/rewards/rewards.go

```diff
@@ -7,79 +7,45 @@ import (
 	"github.com/spacemeshos/go-spacemesh/sql"
 )
 
-const (
-	coinbaseReward = iota + 1
-	smesherReward
-)
-
 // Add reward to the database.
-func Add(db sql.Executor, lid types.LayerID, reward *types.AnyReward) error {
-	if _, err := db.Exec(`insert into rewards 
-			(smesher, coinbase, layer, total_reward, layer_reward) 
-			values (?1, ?2, ?3, ?4, ?5) 
-		on conflict(smesher,layer) 
-			do update set 
-				total_reward=add_uint64(total_reward, ?4),
-		 		layer_reward=add_uint64(layer_reward, ?5);`,
+func Add(db sql.Executor, reward *types.Reward) error {
+	if _, err := db.Exec(`insert into rewards (coinbase, layer, total_reward, layer_reward) values (?1, ?2, ?3, ?4);`,
```

Comment:
it was written with do update so that there is no need for creating a map with coinbases

Reply:
what do you mean by `a map with coinbase`? i don't understand this comment.
in the old code, `do update` loses the info at which layer the reward was given

Reply:
oh i was wrong. it has the on conflict check. sorry.

Reply:
changed it back to the old code. also updated test

Reply:
yes, so the idea was that it is possible to write many rewards for the same layer. so if there is a list with rewards there is no need to create a map with a unique coinbases